engineioxide/src/socket.rs

```diff
@@ -171,14 +181,64 @@ where
             .replace(handle);
     }
 
+    cfg_if! {
+        if #[cfg(all(feature = "v3", feature = "v4"))] {
```

Comment:
Why don't use directly the `#[cfg()]` above the heartbeat_job fn rather than using `cfg_if!` ?

Reply:
Good catch, there's no good reason to do so that I know of. I guess I was just overthinking it at that point. I'll make changes.